### PR TITLE
Simple REDIRECT

### PR DESCRIPTION
**Pull request by kelseyroy**
Adds:
* `HTTPServer.Handlers.Redirect`
* 301 `status_code` and "MOVED PERMANENTLY" `status_message`
* `location:` to paths associated with redirect handler
* `build_location_header/1`
* relevant redirect tests

### CODE REVIEW

**Review by t-keazirian — APPROVED** [review]
Isn't it amazing how you get a little faster and it's a little more manageable to add new functionality once you've started to put together all the building blocks and see it come into a fuller, more complete picture?

File: lib/response.ex [hunk 1]
```diff
@@ -46,11 +47,18 @@ defmodule HTTPServer.Response do
     }
   end
 
+  def build_location_header(path) do
+    %{
+      "Location" => "http://#{@host}#{path}"
+    }
+  end
```
Comment (Aoxian) [inline lib/response.ex:54]: adding a named function for all the hundreds of potential http headers will eventually get tedious. Have you considered the [builder pattern](https://refactoring.guru/design-patterns/builder)?

**Review by ghost — APPROVED** [review]

File: lib/response.ex (hunk 1 above)
Comment (kelseyroy) [reply lib/response.ex:54]: The builder pattern refactor is getting pretty big, so I'm going to make it it's own PR!

**Comment by kelseyroy** [comment]
[Simple Redirect](https://trello.com/c/rdVj9doy/21-simple-redirect)

**Pull request merged**